homeassistant/components/qnap/__init__.py

```diff
@@ -1 +1,97 @@
 """The qnap component."""
+from datetime import timedelta
+import logging
+
+from qnapstats import QNAPStats
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.helpers import config_per_platform
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DEFAULT_PORT, DEFAULT_TIMEOUT, DOMAIN, PLATFORMS
+
+UPDATE_INTERVAL = timedelta(minutes=1)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set up the qnap environment."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Import configuration from sensor platform
+    config_platform = config_per_platform(config, "sensor")
+    for p_type, p_config in config_platform:
+        if p_type != DOMAIN:
+            continue
+
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": config_entries.SOURCE_IMPORT},
+                data=p_config,
+            )
+        )
```

Comment:
The import should be done inside `sensor.py`, inside `setup_platform` where it was previously.

Reply:
Wont this limit us in the future for when we want to expand and add additional platforms?

Reply:
No it won't.
Imports should only be done for "old" platforms (=YAML)
In the future, the new platforms will work only from config entry.

Reply:
It should be fully removed from `__init__` module.
But it is compulsory to add it inside `sensor` module.